src/test/java/org/broadinstitute/consent/http/service/FileStorageObjectServiceTest.java

```diff
@@ -0,0 +1,241 @@
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+public class FileStorageObjectServiceTest {
+
+    @Mock
+    FileStorageObjectDAO fileStorageObjectDAO;
+
+    @Mock
+    GCSService gcsService;
+
+    FileStorageObjectService service;
+
+    @Before
+    public void setUp() {
+        openMocks(this);
+    }
+
+    private void initService() {
+        service = new FileStorageObjectService(fileStorageObjectDAO, gcsService);
+    }
+
+    @Test
+    public void testUploadAndStoreFile() throws IOException {
+        InputStream content = new ByteArrayInputStream(new byte[]{});
+        String fileName = RandomStringUtils.randomAlphabetic(10);
+        String mediaType = RandomStringUtils.randomAlphabetic(10);
+        FileCategory category = List.of(FileCategory.values()).get(new Random().nextInt(FileCategory.values().length));
+        String entityId = RandomStringUtils.randomAlphabetic(10);
+        Integer createUserId = new Random().nextInt();
+
+        String bucket = RandomStringUtils.randomAlphabetic(10);
+        String blob = RandomStringUtils.randomAlphabetic(10);
+
+
+        when(fileStorageObjectDAO.insertNewFile(
+                eq(fileName),
+                eq(category.getValue()),
+                eq(BlobId.of(bucket, blob).toGsUtilUri()),
+                eq(mediaType),
+                eq(entityId),
+                eq(createUserId),
+                any())).thenReturn(10);
+
+        FileStorageObject newFileStorageObject = new FileStorageObject();
+        newFileStorageObject.setFileName(RandomStringUtils.randomAlphabetic(10));
+
+        when(fileStorageObjectDAO.findFileById(10)).thenReturn(newFileStorageObject);
+        when(
+                gcsService.storeDocument(eq(content), eq(mediaType), any())
+        ).thenReturn(BlobId.of(bucket, blob));
+
+        spy(fileStorageObjectDAO);
+        spy(gcsService);
+
+        initService();
+
+        FileStorageObject returned = service.uploadAndStoreFile(content, fileName, mediaType, category, entityId, createUserId);
+
+        assertEquals(newFileStorageObject, returned);
+
+        verify(
+                gcsService, times(1)
+        ).storeDocument(eq(content), eq(mediaType), any());
+
+        verify(
+                fileStorageObjectDAO, times(1)
+        ).insertNewFile(
+                eq(fileName),
+                eq(category.getValue()),
+                eq(BlobId.of(bucket, blob).toGsUtilUri()),
+                eq(mediaType),
+                eq(entityId),
+                eq(createUserId),
+                any());
+
+    }
+
+    @Test
+    public void testFetchById() throws IOException {
+        String bucket = RandomStringUtils.randomAlphabetic(10);
+        String blob = RandomStringUtils.randomAlphabetic(10);
+
+        FileStorageObject file = new FileStorageObject();
+        file.setBlobId(BlobId.of(bucket, blob));
+
+        String content = RandomStringUtils.randomAlphabetic(10);
```

Comment:
Thinking out loud - this could be a case where we want to test content that would NOT be db appropriate (i.e. `RandomStringUtils.random`) since it is file content, not stored metadata.